datafusion/core/src/test/mod.rs

```diff
@@ -173,7 +173,10 @@ pub fn partitioned_file_groups(
             writers[partition].write_all(b"\n").unwrap();
         }
     }
-    for w in writers.iter_mut() {
+
+    // Must drop the stream before creating ObjectMeta below as drop
+    // triggers finish for ZstdEncoder which writes additional data
```

Comment:
We previously got away with this as the requested range was being ignored by LocalFileSystem